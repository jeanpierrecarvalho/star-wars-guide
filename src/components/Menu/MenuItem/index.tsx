import Image from "next/image";
import Link from "next/link";

interface IProps {
  href: string;
  imageSrc: string;
  imageAlt: string;
  title: string;
}

const MenuItem: React.FC<IProps> = ({ href, imageSrc, imageAlt, title }) => {
  return (
    <div
      className="mb-12 hover:scale-105 opacity-90 hover:opacity-100 py-4"
      role="menuitem"
    >
      <Link href={href} passHref aria-label={`Link to ${title}`}>
        <div className="cursor-pointer">
          <div className="flex flex-row justify-center">
            <Image
              src={imageSrc}
              width={300}
              height={300}
              alt={imageAlt}
              priority={true}
            />
          </div>
          <div className="flex flex-row justify-center mt-4">
            <span
              className="font-starjout text-4xl text-center text-yellow-500"
              aria-label={`Title: ${title}`}
            >
              {title}
            </span>
          </div>
        </div>
      </Link>
    </div>
  );
};

export default MenuItem;
