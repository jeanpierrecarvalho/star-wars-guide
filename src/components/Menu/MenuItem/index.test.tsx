import { render } from "@testing-library/react";
import { describe, it, expect } from "vitest";

import MenuItem from "@/components/Menu/MenuItem";

describe("MenuItem Component", () => {
  it("matches the snapshot", () => {
    const props = {
      href: "/some-path",
      imageSrc: "/images/characters.png",
      imageAlt: "Sample image",
      title: "Sample Title",
    };

    const { container } = render(<MenuItem {...props} />);
    expect(container).toMatchSnapshot();
  });
});
