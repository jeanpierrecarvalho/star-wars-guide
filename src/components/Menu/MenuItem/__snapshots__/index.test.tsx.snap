// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MenuItem Component > matches the snapshot 1`] = `
<div>
  <div
    class="mb-12 hover:scale-105 opacity-90 hover:opacity-100 py-4"
    role="menuitem"
  >
    <a
      aria-label="Link to Sample Title"
      href="/some-path"
    >
      <div
        class="cursor-pointer"
      >
        <div
          class="flex flex-row justify-center"
        >
          <img
            alt="Sample image"
            data-nimg="1"
            decoding="async"
            fetchpriority="high"
            height="300"
            src="/_next/image?url=%2Fimages%2Fcharacters.png&w=640&q=75"
            srcset="/_next/image?url=%2Fimages%2Fcharacters.png&w=384&q=75 1x, /_next/image?url=%2Fimages%2Fcharacters.png&w=640&q=75 2x"
            style="color: transparent;"
            width="300"
          />
        </div>
        <div
          class="flex flex-row justify-center mt-4"
        >
          <div
            class="border-2 border-yellow-500 bg-gray-900 px-2 border-opacity-60"
          >
            <span
              aria-label="Title: Sample Title"
              class="font-starjout text-4xl text-center text-yellow-500"
            >
              Sample Title
            </span>
          </div>
        </div>
      </div>
    </a>
  </div>
</div>
`;
